src/pip/_internal/resolution/resolvelib/provider.py

```diff
@@ -75,6 +77,133 @@ def _get_with_identifier(
     return default
 
 
+def _extract_names_from_causes_and_parents(
+    causes: Iterable["PreferenceInformation"],
+) -> Set[str]:
+    """
+    Utility function to extract names from the causes and their parent packages
+
+    :params causes: An iterable of PreferenceInformation
+
+    Returns a set of strings, each representing the name of a requirement or
```

Comment:
```suggestion
    Returns a set of strings, representing the name of a requirement and
```

The code adds both the name and the parent's name if there's a parent. (There may be a better way of re-wording this, I don't think Github's "suggestion" mechanism allows for multi-line changes...)